src/main/java/io/jenkins/docker/client/ListContainersCmdExec.java

```diff
@@ -6,9 +6,9 @@
 import com.github.dockerjava.api.command.ListContainersCmd;
 import com.github.dockerjava.api.model.Container;
 import com.github.dockerjava.core.DockerClientConfig;
-import com.github.dockerjava.netty.MediaType;
-import com.github.dockerjava.netty.WebTarget;
-import com.github.dockerjava.netty.exec.AbstrSyncDockerCmdExec;
+import com.github.dockerjava.core.MediaType;
+import com.github.dockerjava.core.WebTarget;
+import com.github.dockerjava.core.exec.AbstrSyncDockerCmdExec;
```

Comment:
I _suspect_ that, now we're using the official `NettyDockerCmdExecFactory` instead of our own version, there's nothing that references this class either.
The reaon we needed this class was because there was a bug in the `docker-java` code where the Netty implementation (accidentally) depended on a Jersey `FilterEncoder` class which broke it for Jenkins usage.
(see the comments in this file for details)

So we might well be able to delete this class too.

Note: We would need to test that we didn't get nasty exceptions when trying to ask "so, how many containers are we running then?", as that was the bug this class was here to fix.  If that functionality works without this class being used then that'd tell us we don't need this class anymore.

Reply:
Ok, I remove it since it's not used anymore.
Can you give me a simple way to reproduce the bug it was supposed to fix, so I can test it?

Reply:
The bug caused an inability to do a "list instances with these labels" query.  The code uses labels to distinguish between containers it started and anything else running within the docker host.

To test:
1. Configure a single cloud with a single template.
1. Set a cloud instance limit of 2, template instance max of 1.
1. Have a build require that template and trigger the build.

With the bug, when the `DockerCloud` code tries to ask the docker host "how many instances do we have running in total" (method `countContainersInDocker`), there would be a big ClassNotFound exception.
...and when it tries to ask the docker host "how many instances do we have of this particular template", it'd get the same exception.
If you don't set any cloud-instance limits or template instance limits then the code doesn't get called and you're immune to the bug (but can't test for it); with instance-count limits set _and_ the bug present, it'll throw and be unable to provision a new container, and hence be unable to start a slave and execute the build.

If the code works, the build should run ok and you'll see a message (from the `canAddProvisionedSlave` method) logged along the lines of `"Provisioning '{}' number {} (of {}) on '{}'; Total containers: {} (of {})"`, where the numbers "in use" came from the list-containers command result.
If we get that log message, it means that the list-containers command worked.

Reply:
I tested it, the build runs OK and I see the appropriate message in the log.

Reply:
As you've probably noticed, I've merged this PR "as is" because we need the "work with docker-java-api-plugin version 3.1.5" changes _now_ so I'm hoping that it'll all work.
I did some basic smoke-tests locally before merging, and it all seemed ok.
...but then the automated Jenkins-CI build failed for seemingly-unrelated reasons ☹️.  Twice 😠 

Given that the release of docker-java-api-plugin has broken the world, we've got little to lose by doing a release, so that's what I'm doing right now...